### PR TITLE
Issue 186 showError override prop

### PR DESCRIPTION
**Pull request by alexjsdev**
take 2 for this: https://github.com/lookfirst/mui-rff/issues/186

I believe this should cover what we discussed in the issue above, and in my [previous PR](https://github.com/lookfirst/mui-rff/pull/191).

TL;DR this will allow users to override the default `showError` function by either passing their own or one of the two pre-defined options that i exported out of Util. Also updated the readme with a new section regarding how to use this. Let me know if there's anything else needed to merge this.

### CODE REVIEW

**Comment by lookfirst** [comment]
Thanks, I'll have a look soon. Been a bit distracted with everything on this planet. =(

**Comment by alexjsdev** [comment]
@lookfirst any reason this can't be merged?

File: README.md [hunk 1]
```diff
@@ -508,6 +508,40 @@ const required = makeRequired(schema);
...
+## showErrorOnChange({ meta }) - default
+Triggers error messages to show up as soon as a value of a field changes. Useful for when the user needs instant feedback from the form validation (i.e. password creation rules, non-text based inputs like select, or switches etc.)
+
+## showErrorOnBlur({ meta })
+Triggers error messages to render after a field is touched, and blurred (focused out of), this is useful for text fields which might start out erronous but end up valid in the end (i.e. email, or zipcode). In these cases you don't want to rush to show the user a validation error message when they haven't had a chance to finish their entry.
+
+```tsx
+import { TextField, showErrorOnBlur } from 'mui-rff';
+
+<TextField label="Hello world" name="hello" required={true} showError={showErrorOnBlur}/>
```
Comment (mickrip) [inline README.md:526]: This is awesome. Much better. :)

Comment (marr) [reply README.md:526]: Looking forward to this change as well.

**Comment by marr** [comment]
@alexjsdev Is this something we can merge soon? I use mui-rff on my login /signup form and having the ability to validate on blur is crucial!

Thanks

**Comment by alexjsdev** [comment]
@marr I'm in the same ⛵ I don't have merge/publish access to this repo/package so unfortunately it's not up to me, otherwise I would have merged this already. Still waiting for a reply from @lookfirst, it's his lib so he's the one with the power ⚡ 🤷

**Comment by lookfirst** [comment]
I'll be back soon. =) Just finishing up a few more things with work and life and I'll be full steam on mui-rff again soon.

**Comment by lookfirst** [comment]
Thanks @alexjsdev

**Comment by alexjsdev** [comment]
no prob @lookfirst. just saw this reply... any reason this can't be merged? I'm currently building my app for work off of my fork via GitHub repo, but I'd much rather use the real package. please let me know if there are other changes required from my end? thanks

**Comment by lookfirst** [comment]
sorry @alexjsdev ... i tend to get about 30 minutes to work on this project and then get pulled away for a few days... it sucks. i miss it. =( i'll get some time soon hopefully.

**Comment by lookfirst** [comment]
thanks for all the hard work on this @alexjsdev 

i just merged it in and released as v2.3.0

**Comment by alexjsdev** [comment]
sweet! 🎉

**Pull request merged**